### ISSUE TITLE
Amazfit Cor fw 1.2.7.20 problem with the firmware metadata transfer

### ISSUE THREAD

**Issue by bertusbotha83** [Freeyourgadget/Gadgetbridge#1440]
#### Before opening an issue please confirm the following:
- [x] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [x] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [x] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

#### Your issue is:
I am trying to flash my Amazfit Cor since it has been bricked since the last update and unable to pair with the Mi Fit app. It does however pair with the Gadgetbridge app, but fails when I try to flash the firmware. It looks like the firmware flash is happening on the device (the blue circle finishes loading) but at the end I just get a message that says "Problem with the firmware metadata transfer".

#### Your wearable device is:

Amazfir Cor fw V1.2.7.20

#### Your android version is:

8.0.0 (EMUI 8.0.0)

#### Your Gadgetbridge version is:

0.32.3

*New issues about already solved/documented topics could be closed without further comments. Same for too generic or incomplete reports.*
![screenshot_20190221-151430](https://user-images.githubusercontent.com/20242734/53173839-1f4c5f00-35f1-11e9-90c9-db3cd218036e.jpg)
[gadgetbridge.log](https://github.com/Freeyourgadget/Gadgetbridge/files/2889544/gadgetbridge.log)

**Comment by jmvaras2** [comment]
I have the same problem. I've tried to upload also 1.2.7.22 and the problem is the same.

**Comment by ashimokawa** [comment]
Thanks will look into it

**Comment by danielegobbetti** [comment]
I tested it and it worked here. Nevertheless I have only one android device (out of three in my house) that is able to flash **any** firmware to the cor. Can you try with another android device?

**Comment by bertusbotha83** [comment]
Unfortunately, I only have one Android device in my home, the rest is Apple. I did not find a solution to my bricked Amazfit Cor and ultimately gave up and bought myself a new fitness band, Huawei Honor Band 4. If anyone does solve this problem, I would still appreciate hearing the solution.

**Comment by miji2** [comment]
Same problem with same FW version on same hardware for the same reason. Midong bricked after last Mi Fit update and unable to pair with Mi Fit. Used several different mobile devices with same result. Android 7 and 9, same result. Different Mi account, same result.  Gadgetbridge fails firmware flash with "problem with the firmware metadata transfer". Both Gadgetbridge and Notify&Fitness seem to transfer firmware but fail at the end - may be some new verification? Flashing older firmware produces same error.

**Comment by jmvaras2** [comment]
I have changed my phone for a new Redmi Note 7 and when I tried to sync the band it did work. MiFit 3.5.9 updated the firmware of the band to 1.2.7.22 correctly and now it's working. I was thinking on returning it to China, so I'm very happy.

**Comment by miji2** [comment]
Yeah, I saw new firmware so I tried with MiFit 3.5.9 on Mi 5, Mi Max, and a Samsung phone ... can't pair via MiFit but connects via Gadgetbridge. Flashing 1.2.7.22 via Gadgetbridge still produces a "problem with the firmware metadata transfer" no matter which version of firmware I used.  :(

**Comment by jmvaras2** [comment]
It happened the same to me. I was decided to send the band to China because it's on warranty. I also tried with 3 different phones (Redmi Note 3, Galaxy S4 and Galaxy S5) and the result was always the same with Gadgedtfree and with Notify & Fitness for Amazing. Both sent the firmware until the end, but then gave the error. I've bought a new Redmi Note 7 and my surprise was when I tried to connect the band directly to MiFit. It worked. I suspect that it's something related with the Bluetooth connection historic or something similar because I tried it many times before with the other phones I have without success.

**Comment by miji2** [comment]
Here how I was able to work around this finally.
1. remove all pairings within the phone and within Gadgetbridge
2. Pair Gadgetbridge with the Midong band
3. immediately go to Hamburger menu --> Debug, and click "Factory Reset". Band rebooted and reset bluetooth MAC address
4. band started to pair with Mi Fit (and updated to new firmware). Success!
I could have sworn I tried this before, but it actually took effect this time and the band rebooted. It is now fully functional.

**Comment by bertusbotha83** [comment]
Thanks for the update. I managed to unbrick mine as well. I think it also
came down to restarting the band. For you, you did it via "Factory Reset",
but for me, I just waited 2 months for the band to run out of power and
after it restarted it was able to pair again.
